File: aTan/DSU/G/sinh.cpp
#include <bits/stdc++.h>

using namespace std;

mt19937_64 rng(chrono::high_resolution_clock::now().time_since_epoch().count());
int randint(int l, int r) { return uniform_int_distribution<int>(l, r)(rng); }

signed main() {
  system("g++ main.cpp -o main");
  system("g++ trau.cpp -o trau");

  int limTest = 555;
  int lim = 273;

  for (int test = 1; test <= limTest; test++) {
    ofstream inp("test.inp");

    int n = randint(1, lim);
    int m = randint(1, lim);
    inp << n << " " << m << "\n";

    for (int i = 0; i < n; i++) {
      for (int j = 0; j < m; j++) {
        inp << randint(1, lim) << " ";
      }
      inp << "\n";
    }

    int k = randint(1, lim);
    inp << k << "\n";
    for (int i = 0; i < k; i++) {
      inp << randint(1, lim) << "\n";
    }

    inp.close();

    system("./main");
    system("./trau");

    cout << "--- RUNNING TEST: " << test << " ---\n";
    cout << "RESULT: ";
    int check = system("diff testChuan testTrau");
    if (check != 0) {
      cout << "WA!!!\n\n";
      return 0;
    }
    cout << "AC\n\n";
  }

  cout << "ALL " << limTest << " TEST AC - NICE!!! \n\n";
}

File: aTan/DSU/G/sinh.py
from subprocess import run, TimeoutExpired, CalledProcessError
from random import randint


def gen(lim):
    with open("test.inp", "w") as f:
        n = randint(1, lim)
        m = randint(1, lim)

        f.write(f"{n} {m}\n")
        for i in range(n):
            for j in range(m):
                f.write(f"{randint(1,lim)} ")
            f.write("\n")

        k = randint(1, lim)
        f.write(f"{k}\n")
        for i in range(k):
            f.write(f"{randint(1,lim)} ")


def exc(prog, time):
    try:
        run([f"./{prog}"], timeout=time, check=1)
        return 0
    except TimeoutExpired:
        print("[TLE]")
        return 1
    except CalledProcessError:
        print("[RE]")
        return 1


run(["g++", "main.cpp", "-o", "main"])
run(["g++", "trau.cpp", "-o", "trau"])

limTest = 10
lim = 500

for test in range(limTest):
    print(f"TEST {test+ 1}: ", end="")

    gen(lim)
    check = 0
    check = check or exc("main", 1)
    check = check or exc("trau", 1)

    with open("testChuan", "r") as f:
        chuan = f.read()
    with open("testTrau", "r") as f:
        trau = f.read()

    if chuan != trau and not check:
        print("[WA]")
    elif not check:
        print("[AC]")

print("ALL AC")
